src/models/Automaton.java

```diff
@@ -46,6 +51,23 @@ public Automaton(String name, List<Location> locations, List<Edge> edges, List<C
             }
         }
 
+        /* The finite set of actions must be partitioned into inputs and outputs.
+         *   Here we check whether they are partitioned by checking that the intersection of the
+         *   inputs and outputs are empty as an action can only be either an input xor an output.
+         *   We don't have to check whether an action is neither an input nor an output as the set
+         *   of actions is build as the union of the inputs and outputs and for this reason
+         *   guarantees to be in the set of actions. */
+        Set<Channel> intersection = new HashSet<>(inputAct);
+        intersection.retainAll(outputAct);
+        if (!intersection.isEmpty()) {
+            throw new IllegalArgumentException("The action set of a specification must be a partition, where each action is either an input xor an output");
+        }
+
+        /* As the inputs and outputs are shown to be disjoint then the union of them
+         *   it holds that the disjoint union of inputs and outputs is a finite set
+         *   of actions partitioned into inputs and outputs. Here the "disjoint" comes
+         *   from actions (or Channels) are reference types and are for this reason not
+         *   merged with actions of the same name. */
```

Comment:
Comment has some English grammar issues, but I mainly think it's too wordy for what you actually try to convey.

Maybe something like

`Since inputAct and outputAct are now disjoint, we can simply construct actions as the union of both sets.`

Reply:
Very good and comment.